src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -79,35 +81,26 @@ public class ParameterProvider {
 
   public static final boolean ENABLE_NEW_JSON_PARSING_LOGIC_DEFAULT = true;
 
-  public static final boolean IS_ICEBERG_MODE_DEFAULT = false;
+  public static final boolean ENABLE_ICEBERG_STREAMING_DEFAULT = false;
 
   /** Map of parameter name to parameter value. This will be set by client/configure API Call. */
   private final Map<String, Object> parameterMap = new HashMap<>();
 
-  /* Iceberg mode flag */
-  private final boolean isIcebergMode;
-
   // Cached buffer flush interval - avoid parsing each time for quick lookup
   private Long cachedBufferFlushIntervalMs = -1L;
 
+  // Cached enableIcebergStreaming - avoid parsing each time for quick lookup
+  private Boolean cachedEnableIcebergStreaming = null;
```

Comment:
curious, why not primitive boolean that's set to false by default?

Reply:
Use null to represent that is not parsed yet. Once parsed will be cached in the `cachedEnableIcebergStreaming` with either true or false.